src/feditest/protocols/webfinger/__init__.py

```diff
@@ -90,35 +90,31 @@ class WebFingerClient(WebClient):
     """
     A Node that acts as a WebFinger client.
     """
-    def perform_webfinger_query(self, resource_uri: str, rels: list[str] | None = None) -> WebFingerQueryResponse:
+    def perform_webfinger_query(
+        self,
+        server: WebFingerServer,
+        resource_uri: str,
+        rels: list[str] | None = None,
+        scheme: str = 'https'
+    ) -> WebFingerQueryResponse:
         """
-        Make this Node perform a WebFinger query for the provided resource_uri.
+        Make this Node perform a WebFinger query for the provided resource_uri on the provided server
         The resource_uri must be a valid, absolute URI, such as 'acct:foo@bar.com` or
         'https://example.com/aabc' (not escaped).
-        rels is an optional list of 'rel' query parameters
+        rels is an optional list of 'rel' query parameters. The scheme can be overridden to 'http'
+        for development scenarios
         Return the result of the query. This should returns WebFingerQueryResponse in as many cases
         as possible, but the WebFingerQueryResponse may indicate errors.
         """
         raise NotImplementedByNodeError(self, WebFingerClient.perform_webfinger_query)
 
-    def construct_webfinger_query_for(
-        self,
-        server_prefix: str,
-        resource_uri: str,
-        rels: list[str] | None = None,
-    ) -> str:
-        query = (
-            f"{server_prefix}/.well-known/webfinger?resource={quote(resource_uri)}"
```

Comment:
This isn't equivalent to the hostname based string created by `construct_webfinger_uri_for` (even with the new `scheme` argument). The server prefix may also include a port. What about changing `construct_webfinger_uri_for` to create a server prefix (assuming default port numbers) and call `construct_webfinger_query_for`? Would that work?

Reply:
I've lost track on the exact use case for prefix/scheme/port. I think it goes back to me wanting to be able to run things in a development environment, but we left it vague what exactly that means... I'm worried about 1) consistency: if we allow these variations here, where else should we also allow them? The APIs should look the same 2) Can a developer understand what a "server-prefix" is without lengthy explanations?

Reply:
I think a typical developer wouldn't need a lengthy explanation. The server prefix is just the scheme, host and optional port. If you think the name is the confusing, we could call it base_url or something like that.

Reply:
What about this. We freeze this as-is, and pick up the conversation again when we have an actual situation where somebody wants to do something like this.

Reply:
By "something like this", do you mean using the server prefix? If so, many of the tests are already using it (originally, to support the http vs https feature). I used the custom port number for some local development testing. Maybe we should chat about it?

Reply:
My preference is to support server_prefix since it combines scheme, hostname properties and supports a custom port. However, in the spirit of making progress, go ahead and merge the current changes if you disagree.

Reply:
I wasn't aware that you were running it with a custom port. Actually could you do #323 for your current setup? I assume the setup is sufficiently useful/general others might also want to do it?

Reply:
I commented on #323.